practice/jokes/jokes/models.py

```diff
@@ -0,0 +1,54 @@
+"""Jokes models."""
+
+
+class Joke:
+    """
+    Value type that represents a submitted joke. It has a setup, and a punchline.
+
+    >>> a = Joke('words', 'wordswords')
+    >>> a
+    'Joke(words, wordswords)'
+    """
+    def __init__(self, setup, punchline):
+        self.setup = setup
+        self.punchline = punchline
+
+    def __eq__(self, other):
+        return (
+            self.setup == other.setup and
+            self.punchline == other.punchline
+        )
+
+    def __repr__(self):
+        return 'Joke({!r}, {!r})'.format(self.setup, self.punchline)
+
+jokes = []
+
+def add_joke(setup, punchline):
+    """
+    Takes in an argument of a setup and a punchline, and creates a new Joke class object, which is then added to the
+    global array of jokes.
+
+    >>> add_joke('words', 'wordswords')
+    'Joke(words, wordswords)'
```

Comment:
This doctest doesn't pass, also because this function uses globals, it's kind of untestable. Try using the [functional core, state-ful shell](https://github.com/PdxCodeGuild/Full-Stack-Day-Class/blob/master/sessions/2016-07-07/tips/2016-08-31.md#functional-core-stateful-shell) trick to make it more testable.